src/meta/src/hummock/manager/compaction/mod.rs

```diff
@@ -728,6 +729,22 @@ impl HummockManager {
                 }
             }
 
+            let table_watermarks = version
+                .latest_version()
+                .table_watermarks
+                .iter()
+                .filter_map(|(table_id, table_watermarks)| {
+                    if matches!(
+                        table_watermarks.watermark_type,
+                        WatermarkSerdeType::PkPrefix,
+                    ) {
+                        Some((*table_id, table_watermarks.clone()))
+                    } else {
+                        None
+                    }
+                })
+                .collect();
```

Comment:
Actually why don't we do the filtering inside the picker instead like in here if the watermark type is part of TableWatermarks: https://github.com/risingwavelabs/risingwave/blob/5ed4920f2c61bd4a1ebf25ec902b0c9521d01c15/src/meta/src/hummock/compaction/selector/vnode_watermark_selector.rs#L53

We can avoid cloning the table watermark, which can be large given that it stores bytes from user data, with no harm.